src/ir/auth_logic/souffle_emitter.h

```diff
@@ -61,6 +68,17 @@ class SouffleEmitter {
   }
 
   std::string EmitPredicate(const datalog::Predicate& predicate) {
+    const absl::string_view comparison_operators[] = {"<",  ">",  "=",
+                                                      "!=", "<=", ">="};
+    // Rvalue rules which are represented as datalog::Predicates in AST, we use
+    // infix notation while translating to datalog.
+    // Ex: "<"(number1, number2) -> number1 < number2.
+    if (absl::c_find(comparison_operators, predicate.name()) !=
```

Comment:
When you change `comparison_operators` to a set, 
```C++
if (comparison_operator.count(predicate.name()) > 0) {
```

This avoids a linear search as in the other cases.